### DISCUSSION TITLE
Is session and caching supported.

### DISCUSSION THREAD

**Discussion by saisaigraph** [reflex-dev/reflex#326] [General]
Session and caching supported.

**Comment by Alek99** [comment]
We have a state for each users session is that what you are referring too. Some of our components can be memoized for components aswell, what do you mean by caching, caching the users state?

**Reply by Alek99** [reply]
I think I see what you are asking. The concepts of caching is from frameworks such as streamlit. But since we compile down to a nextjs/react app they are not needed. This is because frameworks like streamlit render your ui on every state change so you need concepts such as caching to speed this up because there framework is inherently slow. In Pynecone our UI gets compiled to a nextjs/react app no nothing needs to be cached its way faster and doesn't suffer the draw backs streamlit like frameworks have

**Reply by saisaigraph** [reply]
Wow Amazing ... guys... Thanks Alek... yes i m coming from streamlit background...

**Reply by Alek99** [reply]
No problem thanks for checking out Pynecone! Yeah that was the main reason we made Pynecone, we were frustrated by the performance and lack of customizability streamlit and other frameworks like it provided.
